tests/integration/test_heartbeats.py

```diff
@@ -116,7 +116,10 @@ def verify_user_agent(user_agent, expected, envname=None, marker=None):
             # It also has the advantage of making sure our code respects proxies
             # fmt: off
             cmd = ["proxyspy", "--return-code", "404"]
-            cmd.extend(["--delay", "2.0" if hval == "delay" else "0.1"])
+            cmd.extend(("--prepare-host", "repo.anaconda.com"))
+            cmd.extend(("--prepare-host", "conda.anaconda.org"))
+            if hval == "delay":
```

Comment:
By using the new `--prepare-host` capability in proxyspy we can pre-compute the SSL certificates for known hosts in advance, so that there are no connection delays caused by dynamic generation.